apps/src/lib/wallet/store.rs

```diff
@@ -49,6 +49,8 @@ pub struct ValidatorData {
 
 #[derive(Serialize, Deserialize, Debug, Default)]
 pub struct Store {
+    /// The validator's address, if the wallet belongs to a validator
```

Comment:
In general, I really don't like types that allow non-meaningful instances. We want the address and keys to both be Some or both be None and nothing in type system enforces that now. Do we really need to do this because we want to be able to set things incrementally?

Reply:
For genesis setup, validators use their keys, but the addresses are generated only once the genesis is being finalized, but we could use a different backing store for this "pre-genesis" wallet (I had it in my history, but I didn't get to finish it):

```rust
/// Validator pre-genesis includes all the required keys for genesis.
#[derive(Serialize, Deserialize, Debug)]
pub struct ValidatorPreGenesis {
    /// Cryptographic keypair for validator account key
    account_key: wallet::StoredKeypair,
    /// Cryptographic keypair for consensus key
    consensus_key: wallet::StoredKeypair,
    /// Cryptographic keypair for rewards key
    rewards_key: wallet::StoredKeypair,
    /// Cryptographic keypair for Tendermint node key
    tendermint_node_key: wallet::StoredKeypair,
    /// Special validator keys
    pub(crate) validator_keys: wallet::ValidatorKeys,
}
```

It's a good point and it will actually simplify error handling for the genesis validator setup a lot, so I'm in favor of fixing this before merging

Reply:
Thanks for this suggestion! I dropped the commit that split up the validator address from validator keys and added the new pre-genesis wallet in 415a579 and eabe7e6